src/main/scala/org/apache/spark/deploy/armada/submit/ConfigGenerator.scala

```diff
@@ -43,8 +44,12 @@ private[submit] class ConfigGenerator(val prefix: String, val conf: SparkConf) {
 
   // Store the config files as annotations in the armada submit request
   def getAnnotations: Map[String, String] = {
-    confFiles.map(f =>
-      (prefix + "/" + f.getName, Source.fromFile(f.toString).mkString)).toMap
+    confFiles
+      .map(f => prefix + "/" + f.getName -> f).toMap
+      .view.mapValues(f =>
+        Using(Source.fromFile(f.toString)) { source => source.mkString }.get
```

Comment:
`Using` makes sure the closable object returned by `Source.fromFile(f.toString)` is always being closed (`try { } finally { }`).

It returns a `Try` object, which is a class construct for `try { } catch { }`, providing programmatic access to the result or exception. With `Try.get`, the value is returned, or the exception is thrown.